app/src/docs/v1.api-spec.yaml

```diff
@@ -344,6 +344,68 @@ paths:
           $ref: "#/components/responses/Forbidden"
         default:
           $ref: "#/components/responses/Error"
+  /object/{objId}/metadata:
+    patch:
+      summary: Adds metadata to an object
+      description: >-
+        Creates a copy and new version of the object with the given metadata
+        added to the object. Multiple Key/Value pairs can be provided in the header for the metadata.
+      operationId: addMetadata
+      tags:
+        - Object
+      parameters:
+        - $ref: "#/components/parameters/Path-ObjectId"
+        - $ref: "#/components/parameters/Query-VersionId"
+        - $ref: "#/components/parameters/Header-Metadata"
+      responses:
+        "204": 
+          description: The resource was added successfully.
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        default:
+          $ref: "#/components/responses/Error"
+    put:
+      summary: Replaces metadata of an object
+      description: >-
+        Creates a copy and new version of the object with the given metadata replacing the existing. Multiple Key/Value pairs can be provided in the header for the metadata.
+      operationId: replaceMetadata
+      tags:
+        - Object
+      parameters:
+        - $ref: "#/components/parameters/Path-ObjectId"
+        - $ref: "#/components/parameters/Query-VersionId"
+        - $ref: "#/components/parameters/Header-Metadata"
+      responses:
+        "204": 
+          description: The resource was added successfully.
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        default:
+          $ref: "#/components/responses/Error"
+    delete:
+      summary: Delete metadata of an object.
+      description: >-
+        Creates a copy and new version of the object with the given metadata removed. Multiple Key/Value pairs can be provided in the header for the metadata. If no metadata headers are given then all metadata will be removed. 
+      operationId: deleteMetadata
+      tags:
+        - Object
+      parameters:
+        - $ref: "#/components/parameters/Path-ObjectId"
+        - $ref: "#/components/parameters/Query-VersionId"
+      responses:
+        "204": 
+          description: The resource was deleted successfully.
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        default:
```

Comment:
If we're going to be doing 422 rejections, that will also need to be defined here as potential HTTP response type.